### ISSUE TITLE
Unable to determine terminal size when redirecting output

### ISSUE THREAD

**Issue by Zebralt** [Zebralt/ctxb#3]
The current way of retrieving terminal size (width, height) is the function `os.get_terminal_size`, which is reasonably time-efficient, enough to not hamper I/O operations :

```python
In [18]: %timeit os.get_terminal_size()
558 ns ± 8.95 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

However, this function breaks when redirecting the output of the python script :

```sh
python myscript.py > out.log
OSError: [Errno 25] Inappropriate ioctl for device

# Same problem with tee
python myscript.py | tee out.log
OSError: [Errno 25] Inappropriate ioctl for device
```

One first idea would be to have the enter/exit behave differently when redirecting, since we won't be seeing the output on screen. However, tools like `tee` allow for such a thing.

We can provide a default terminal size, as `shutil.get_terminal_size` does in case `os.get_terminal_size` fails, but we can't be sure where to write the context bar since we don't know the true width or height of the terminal.

Need to confirm is providing a position outside of the terminal on the y axis still does write at the lowest available position with the related escape sequence `'\033[%d;%dH' % (y, x)`.

**Comment by Zebralt** [comment]
I was dumb enough to read the entire post :
https://stackoverflow.com/questions/566746/how-to-get-linux-console-window-width-in-python/41864359#41864359

Just use `os.get_terminal_size(0)` if the regular call breaks.

**Comment by Zebralt** [comment]
Fixed with 7b5caaa.
